baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/PreferGuavaVisibleForTesting.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+// modeled off of
+// https://github.com/google/error-prone/blob/master/core/src/main/java/com/google/errorprone/bugpatterns/checkreturnvalue/UsingJsr305CheckReturnValue.java
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.ImportTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.suppliers.Supplier;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ImportTree;
+import com.sun.tools.javac.code.Type;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Checker that recommends using Guava's version of {@code @VisibleForTesting} over other copies.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(summary = "Prefer Guava's @VisibleForTesting over other versions.", severity = SeverityLevel.WARNING)
+public final class PreferGuavaVisibleForTesting extends BugChecker implements ImportTreeMatcher {
```

Comment:
These are Error-Prone annotations, not Guava

```suggestion
public final class PreferErrorProneAnnotations extends BugChecker implements ImportTreeMatcher {
```

WDYT about reframing/restructuring this as a general `PreferErrorProneAnnotations` check so we can add other annotations here in the future?

Reply:
I'm pretty sure that `VisibleForTesting` exists in standard guava without error-prone annotation apis.

Reply:
[`@VisibleForTesting` has been in standard Guava since was renamed from Google-collections in 2010.
](https://github.com/google/guava/blame/f5ad01f0326405c321a4d1a7bbafb01b7d4d5074/src/com/google/common/annotations/VisibleForTesting.java)


We may want to consider something like `PreferCommonAnnotations` to steer folks to preferred annotations like cases like this, nullness, etc.

We'll likely be deconflicting between [checker-framework](https://checkerframework.org/), 
[jspecify](http://jspecify.org/docs/start-here/), guava, legacy jsr305 & findbugs/spotbugs, etc. and ideally consolidating on the smallest combination of these.

Reply:
Ah sorry, I didn't look closely enough here. I thought this was for ann EP annotation.

Reply:
Changed to `PreferCommonAnnotations`.  If there are other annotations like VisibleForTesting we'd like to add now, please lmk.  I don't know of others, and am happy to get just this one in.